pytorch_blade/pytorch_blade/torch-mlir/lib/Conversion/TorchToMhlo/DiscTorchToMhlo.cpp

```diff
@@ -274,33 +274,158 @@ class ConvertAtenExtractOp : public OpConversionPattern<AtenOpT> {
+    if (elemTy.isInteger(1)) {
+      if (outTy.isa<IntegerType>() && bitExtend) {
+        rewriter.replaceOpWithNewOp<arith::ExtUIOp>(op, outTy, extractOutput);
+        return success();
+      } else if (arith::UIToFPOp::areCastCompatible(elemTy, outTy)) {
+        rewriter.replaceOpWithNewOp<arith::UIToFPOp>(op, outTy, extractOutput);
+        return success();
+      }
+    }
+
+    if (elemTy.isUnsignedInteger() && outTy.isa<FloatType>()) {
+      // Unsigned integers need an unrealized cast so that they can be passed
+      // to UIToFP.
+      ValueRange valueRange{extractOutput};
+      Value unrealizedCast =
+          rewriter
+              .create<UnrealizedConversionCastOp>(
+                  loc,
+                  rewriter.getIntegerType(elemTy.getIntOrFloatBitWidth()),
+                  valueRange)
+              .getResult(0);
+      rewriter.replaceOpWithNewOp<arith::UIToFPOp>(op, outTy, unrealizedCast);
+      return success();
+    }
+
+    if (arith::SIToFPOp::areCastCompatible(elemTy, outTy))
+    // All other si-to-fp conversions should be handled by SIToFP.
+    {
+      rewriter.replaceOpWithNewOp<arith::SIToFPOp>(op, outTy, extractOutput);
+      return success();
+    }
+
+    // integer to integer
+    if (elemTy.isa<IntegerType>() && outTy.isInteger(1)) {
+      // Casting to boolean, integers need to only be checked as not-equal to
+      // zero.
+      Value zero = rewriter.create<arith::ConstantIntOp>(
+          loc, 0, elemTy.getIntOrFloatBitWidth());
+      rewriter.replaceOpWithNewOp<arith::CmpIOp>(
+          op, arith::CmpIPredicate::ne, extractOutput, zero);
+      return success();
+    }
+
+    if (elemTy.isa<IntegerType>() && outTy.isa<IntegerType>() && bitExtend) {
+      rewriter.replaceOpWithNewOp<arith::ExtSIOp>(op, outTy, extractOutput);
+      return success();
+    }
+
+    if (elemTy.isa<IntegerType>() && outTy.isa<IntegerType>() && !bitExtend) {
+      auto intMin = rewriter.create<arith::ConstantIntOp>(
+          loc,
+          APInt::getSignedMinValue(outTy.getIntOrFloatBitWidth())
+              .getSExtValue(),
+          elemTy.getIntOrFloatBitWidth());
+
+      auto intMax = rewriter.create<arith::ConstantIntOp>(
+          loc,
+          APInt::getSignedMaxValue(outTy.getIntOrFloatBitWidth())
+              .getSExtValue(),
+          elemTy.getIntOrFloatBitWidth());
+
+      auto smallerThanMin = rewriter.create<arith::CmpIOp>(
+          loc, arith::CmpIPredicate::slt, extractOutput, intMin);
+      auto minOrArg = rewriter.create<arith::SelectOp>(
+          loc, smallerThanMin, intMin, extractOutput);
+      auto largerThanMax = rewriter.create<arith::CmpIOp>(
+          loc, arith::CmpIPredicate::slt, intMax, extractOutput);
+
+      Value clamped = rewriter.create<arith::SelectOp>(
+          loc, largerThanMax, intMax, minOrArg);
+      rewriter.replaceOpWithNewOp<arith::TruncIOp>(op, outTy, clamped);
+      return success();
```

Comment:
Nit: Add some e2e unit tests for these complicate conversions

Reply:
There are some troubles to extract low bit width less than torch.int64 in UT. So maybe test this later.

Reply:
I mean e2e ut but not filecheck